core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java

```diff
@@ -84,7 +84,20 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
              // For mutation data only we can have multiple files with the same genetic_profile.
              // There is a constraint in the mutation database table to prevent duplicated data
              // If this constraint is hit (mistakenly importing the same maf twice) MySqlBulkLoader will throw an exception
-             return existingGeneticProfile;
+
+             // make an object combining the pre-existing profile with the file-specific properties of the current file
+             GeneticProfile gp = new GeneticProfile(
+                     existingGeneticProfile.getStableId(),
+                     existingGeneticProfile.getCancerStudyId(),
+                     existingGeneticProfile.getGeneticAlterationType(),
+                     existingGeneticProfile.getDatatype(),
+                     existingGeneticProfile.getProfileName(),
+                     existingGeneticProfile.getProfileDescription(),
+                     existingGeneticProfile.showProfileInAnalysisTab());
+             gp.setGeneticProfileId(existingGeneticProfile.getGeneticProfileId());
+             gp.setTargetLine(geneticProfile.getTargetLine());
+             gp.setOtherMetadataFields(geneticProfile.getAllOtherMetadataFields());
+             return gp;
```

Comment:
`existingGeneticProfile` is part of a static dictionary in `DaoGeneticProfile`, which forms a naive cache of the database. I think that fields specific to the data loading phase of a particular data file in this profile, which are not in the database, do not belong in the cache either. That is why I make a new object to combine the profile information from the cache with the file-specific information from the new meta file.